src/coreclr/vm/interpexec.cpp

```diff
@@ -83,6 +88,108 @@ static void InterpBreakpoint()
 #define LOCAL_VAR(offset,type) (*LOCAL_VAR_ADDR(offset, type))
 #define NULL_CHECK(o) do { if ((o) == NULL) { COMPlusThrow(kNullReferenceException); } } while (0)
 
+template <typename THelper> static THelper GetPossiblyIndirectHelper(void* dataItem)
+{
+    size_t helperDirectOrIndirect = (size_t)dataItem;
+    if (helperDirectOrIndirect & INTERP_INDIRECT_HELPER_TAG)
+        return *(THelper *)(helperDirectOrIndirect & ~INTERP_INDIRECT_HELPER_TAG);
+    else
+        return (THelper)helperDirectOrIndirect;
+}
+
+template <typename TResult, typename TSource> static void ConvFpHelper(int8_t *stack, const int32_t *ip)
+{
+    static_assert(!std::numeric_limits<TSource>::is_integer, "ConvFpHelper is only for use on floats and doubles");
+    static_assert(std::numeric_limits<TResult>::is_integer, "ConvFpHelper is only for use on floats and doubles to be converted to integers");
+
+    // First, promote the source value to double
+    double src = LOCAL_VAR(ip[2], TSource),
+        minValue = (double)std::numeric_limits<TResult>::lowest(),
+        maxValue = (double)std::numeric_limits<TResult>::max();
+
+    // (src != src) checks for NaN, then we check whether the min and max values (as represented by their closest double)
+    //  properly bound the source value so that when it is truncated it will be in range
+    // We assume that we are in round-towards-zero mode. For NaN we want to return 0, and for out of range values, saturate.
+    TResult result;
+    if (src != src)
+        result = 0;
+    else if (src >= maxValue)
+        result = std::numeric_limits<TResult>::max();
+    else if (std::numeric_limits<TResult>::is_signed && (src <= -1))
+        result = 0;
+    else if (src < minValue)
```

Comment:
```suggestion
    else if (!std::numeric_limits<TResult>::is_signed && (src < minValue))
```
?

I am not sure whether the compiler is smart enough to optimize out the unnecessary comparison